src/app/components/vertical-bar-chart/vertical-bar-chart.component.ts

```diff
@@ -0,0 +1,198 @@
+      changes['barChartDataSets'].currentValue
+    ) {
+      this.cleanLegendCheckBox();
+      this.barChartDataSets = changes['barChartDataSets'].currentValue;
+      this.renderChart(this.bucketList, this.barChartDataSets);
+    }
+  }
+
+  renderChart(bucketList: string[], barChartDataSets: BarChartDataSet[]) {
+    if (!this.chartContainerId) {
+      return;
+    }
+    const ctx = (document.getElementById(this.chartContainerId) as HTMLCanvasElement).getContext(
+      '2d'
+    );
+
+    if (this.barChart) {
+      this.barChart.destroy();
+    }
+
+    this.handleLegendCheckBox()
+    this.barChart = new Chart(ctx!, {
+      type: 'bar',
+      data: {
+        labels: this.bucketList,
+        datasets: barChartDataSets.map((item, index) => {
+          return {
+            label: item.label,
+            data: item.data,
+            backgroundColor: item.backgroundColor,
+            borderWidth: item.borderWidth,
+            hidden: this.hiddenDatasets[index],
+          }
+        })
+      },
+      options: {
+        responsive: true,
+        plugins: {
+          legend: {
+            display: false,
+          },
+          title: {
+            display: false,
+          },
+          tooltip: {
+            enabled: true,
+            position: 'nearest',
+            callbacks: {
+              label: function (context) {
+                return barChartDataSets[context.datasetIndex].label;
+              },
+              footer: function (context) {
+                // show bar description on tooltip footer
+                let datasetIndex = context[0].datasetIndex;
+                let dataIndex = context[0].dataIndex;
+                let nodeCount = context[0].parsed.y
+                let unit = nodeCount > 1 ? 'nodes' : 'node';
+                return "Total: " + nodeCount + " " + unit + "\n\n" + barChartDataSets[datasetIndex].description[dataIndex];
+              }
+            }
+          },
+        },
+        scales: {
+          y: {
+            ticks: {
+              stepSize: 1,
+              precision: 0
+            }
+          }
+        }
+      },
+    });
+
+    this.barChart.update();
+  }
+
+  handleLegendCheckBox() {
+    // create checkbox if no HTMLElement in legend div
+    let chartLegendDiv = document.getElementById('chart-legend-div');
+    if (chartLegendDiv) {
+      if (chartLegendDiv.children.length === 0 && this.barChartDataSets.length > 0) {
+        // Record the hidden state of each dataset, by default, the first dataset is visible
+        this.barChartDataSets.forEach((dataset, i) => {
+          this.hiddenDatasets.push(i === 0 ? false : true);
+        });
+
+        chartLegendDiv.appendChild(document.createTextNode('Resource sorted by load:'));
+        chartLegendDiv.appendChild(document.createElement('br'));
+        chartLegendDiv.appendChild(document.createTextNode('(high to low)'));
+        chartLegendDiv.appendChild(document.createElement('br'));
```

Comment:
nit: Resource sorted by load: -> Top 10 resources sorted by load (high to low)

Reply:
Just updated the description.  Thanks for the feedback.